sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -323,6 +323,133 @@ pi_result OCL(piextGetDeviceFunctionPointer)(pi_device device,
                                   function_pointer_ret));
 }
 
+pi_result OCL(piContextCreate)(
+    const cl_context_properties *properties, // TODO: untie from OpenCL
+    pi_uint32 num_devices, const pi_device *devices,
+    void (*pfn_notify)(const char *errinfo, const void *private_info, size_t cb,
+                       void *user_data1),
+    void *user_data, pi_context *retcontext) {
+  pi_result ret = PI_INVALID_OPERATION;
```

Comment:
Nitpicking: Most OpenCL API functions do not return `CL_INVALID_OPERATION`. While the CL API call should always provide an expected return value it might still be worth to initialize the return code variable to something that always works in OpenCL - even if not very helpful - for example `CL_OUT_OF_RESOURCES`?

(Applies to all changes of the create functions).

Reply:
I think it is in opposite good to return something that underlying run-times to not normally return like PI_INVALID_OPERATION, for cases when the run-time did not want (or could not) set its error.

Reply:
We need to document this in pi.h file where we have the declarations of the APIs.